app/helpers/posts_helper.rb

```diff
@@ -0,0 +1,11 @@
+module PostsHelper
+
+  private 
+
+  def logged_in_user
+    unless logged_in? 
+      flash[:danger] = "You must log in in order to submit a post."
+      redirect_to login_url 
+    end 
+  end 
```

Comment:
The general purpose of modules that live in `app/helpers` is to define helpful methods that can be used in your view layer. While it is possible to include them in your controllers (as you've done here), these methods are also available to be used in your erb templates because of their file location, despite the fact that they will never be used there.

For methods like this one, as well as those in your `SessionsHelper`, I'd suggest moving simply defining them in your `ApplicationController`. Since all controllers inherit from that, the methods will be available in all controllers. If then you need to use one of these methods in your views (like `current_user`), you can specify methods to be made available to your views with `helper_method`:

```rb
class ApplicationController < ActionController::Base
  ...

  private

  helper_method :logged_in?, :current_user

  def logged_in?
    !current_user.nil?
  end

  def current_user
    @current_user ||= User.find_by(id: session[:user_id])
  end
end
```